### DISCUSSION TITLE
why is there a copy & paste functionality on the nav layer?

### DISCUSSION THREAD

**Discussion by idavydov** [manna-harbour/miryoku#319] [Q&A] [answered]
I've been using Miryoku for a while (awesome project, thanks!).

However I never made use of the copy & paste functionality on the nav layer. To me homerow mod  ctrl + c/v seems more natural and in QMK it works even if you swap ctrl & gui keys between linux and mac os.

I'm now wondering what's the main reason for having them?

Is it mainly for users who are on dvorak layout? Or is there something else I'm missing?

**Comment by manna-harbour** [comment, marked as answer]
The original motivation was to deal with the different clipboard bindings between applications, in particular between emacs, terminal emulators and everything else. The idea being to use the literal cut / copy / paste / undo / redo keycodes that generally aren't used by applications at all, and translate them into the application-specific bindings per active application, using the window manager or an external utility. That would deal with the issue, but depends on having that utility configured and running. Later I learned of the original CUA clipboard bindings, and they work for emacs, terminal emulators and everything else, but only for cut / copy / paste. So that's now the default, with undo / redo left as direct keycodes requiring remapping in the applications, as there is no CUA binding for those. The original arrangement is still available as an option, as well as an option for modern Windows and Mac bindings.

The other reasons to use them are to reduce layer switching while editing, to help with transition to non-QWERTY alphas, and for consistency with the Mouse and Button layers, including one handed clipboard on Button.
